src/core/public/keyboard_shortcuts/types.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+// Phase 1.1 Core Type Definitions
+
+export interface KeyboardShortcutsSetup {
+  register(shortcuts: ShortcutDefinition[]): string[];
+}
+
+export interface KeyboardShortcutsStart {
+  register(shortcuts: ShortcutDefinition[]): string[];
+  unregister(id: string): void;
+}
+
+export interface ShortcutDefinition {
+  id: string;
+  pluginId: string;
+  name: string;
+  category: string;
+  keys: string;
+  execute: ShortcutHandler;
+}
```

Comment:
lets add comments using JS docs to these, see: https://github.com/angle943/OpenSearch-Dashboards/blob/ef3a9c07a07ffb7f7bfa662d29c88ed487b7e187/src/core/types/app_category.ts